pulsar-broker/src/main/java/org/apache/pulsar/broker/rest/TopicsBase.java

```diff
@@ -433,7 +433,8 @@ private CompletableFuture<Void> lookUpBrokerForTopic(TopicName partitionedTopicN
             }
 
             LookupResult result = optionalResult.get();
-            if (result.getLookupData().getHttpUrl().equals(pulsar().getWebServiceAddress())) {
+            if (result.getLookupData().getHttpUrl().equals(pulsar().getWebServiceAddress())
+                    || result.getLookupData().getHttpUrlTls().equals(pulsar().getWebServiceAddressTls())) {
```

Comment:
Are you able to add a test to cover this behavior?

Reply:
@michaeljmarshall The issue I encounter when trying to help with adding a test case is that it's not clear how to prepare a correct setup for TLS REST client.

I try to copy  `org.apache.pulsar.broker.admin.TopicsTest#testProduceToPartitionedTopic` and change the init logic as:

```java
    @Override
    protected void doInitConf() throws Exception {
        super.doInitConf();
        this.conf.setBrokerServicePort(Optional.empty());
        this.conf.setWebServicePort(Optional.empty());
    }
```

But then the test case failed with SSL HANDSHAKE error or so.

Given that @maanders-tibco is a first-time contribution I wonder if you can share some backgrounds here for test cases we can refer to. Digging into the heavily mocked test suite can be quite disappointing.

Reply:
Sure that makes sense. I'll take a look at the code a bit closer to make sure I understand the change and then we can merge it.